app/src/main/java/de/danoeh/antennapod/fragment/QueueFragment.java

```diff
@@ -340,10 +340,10 @@ public void onConfirmButtonPressed(
                         SortOrder sortOrder = UserPreferences.getQueueKeepSortedOrder();
                         DBWriter.reorderQueue(sortOrder, true);
                         if (recyclerAdapter != null) {
-                            recyclerAdapter.setLocked(true);
+                            recyclerAdapter.setKeepSorted(true);
```

Comment:
I think that could lead to problems when setting it to not sorted again. In that case, it will not be reset. What about using one single method for the adapter (`setDragDropEnabled`)?

Reply:
After merging the methods a bug occurred when i locked the list then enabled and disabled "Keep sorted", when doing that the list would still have drag and drop actions even when blocked solved that by change `setDragDropEnabled` to `updateDragDropEnabled`, the new method looks into the User Preferences instead of the parameters passed. 
Also found a minor bug when the queue is first locked than the "Keep sorted" option is enabled, in that case there is no way to unlock the queue without first disabling "Keep sorted", I thought two options to solve that:
1 - Hiding the "keep sorted" option when lock is enabled
2 - Unlock the queue when "keep sorted" is marked